src/librawspeed/decompressors/LJpegDecompressor.cpp

```diff
@@ -194,15 +194,47 @@ template <int N_COMP, bool WeirdWidth> void LJpegDecompressor::decodeN() {
     copy_n(predNext, N_COMP, pred.data());
     // the predictor for the next line is the start of this line
     predNext = &img(row, col);
+    // the predictor mode is always horizontal on the first line
+    uint32_t predMode = row == 0 ? 1 : predictorMode;
 
     // FIXME: predictor may have value outside of the uint16_t.
     // https://github.com/darktable-org/rawspeed/issues/175
 
     // For x, we first process all full pixel blocks within the image buffer ...
     for (; col < N_COMP * fullBlocks; col += N_COMP) {
       for (int i = 0; i != N_COMP; ++i) {
-        pred[i] = uint16_t(pred[i] + ht[i]->decodeDifference(bitStream));
-        img(row, col + i) = pred[i];
+        img(row, col + i) =
+            uint16_t(pred[i] + ht[i]->decodeDifference(bitStream));
+        if (col < N_COMP * (fullBlocks - 1)) {
+          int predA = img(row, col + i);
+          int predB = predMode > 1 ? img(row - 1, col + N_COMP + i) : 0;
+          int predC = predMode > 1 ? img(row - 1, col + i) : 0;
```

Comment:
FWIW, these are out-of-bounds reads for first row and/or last column.

Reply:
Yep, hence all the checks around...